src/native/libs/System.Native/pal_random.lib.js

```diff
@@ -28,13 +28,17 @@ const DotNetEntropyLib = {
             }
         }
     },
-    dotnet_browser_entropy: function (buffer, bufferLength) {
+
+    // TODO-LLVM: Mono uses a typescript file for mono_wasm_browser_entropy and we probably  don't want to introduce typescript as a dependency
+    // so we stay with the plain Javascript implementation but rename to avoid changing random.c
+    // Revist if and when the Wasi HTTP or CLI proposal progresses (which contain a secure RNG in wasi-random)
```

Comment:
For my education: are browsers expected to implement `wasi-random`?

(P1 of WASI I believe already has some facility for getting these)

Reply:
In my head, yes!  Let me see if I can get something better for educational purposes.

Reply:
I've not been able to find anything about this, nor get any response from a browser team.  I wouldn't be suprised if it happens, it would make sense to me.  However as Typescript is going to be installed for Mono users anyway, perhaps it is only a small ask to NAOT-LLVM to make sure that workload is available also.  Happy to change to the typescript version (and add whatever that means to our build targets) if you'd prefer?

Reply:
No, I think it's completely fine to leave as-is for now. We may revisit this depending on how #2434 goes.

> I've not been able to find anything about this, nor get any response from a browser team. I wouldn't be suprised if it happens, it would make sense to me. 

Thank you for looking into this!